packages/shared/src/hooks/profile/useProfile.ts

```diff
@@ -12,14 +13,11 @@ export function useProfile(initialUser?: PublicProfile): PublicProfile {
     generateQueryKey(RequestKey.Profile, initialUser),
     () => getProfile(initialUser?.id),
     {
-      placeholderData: initialUser,
```

Comment:
For some reason using `placeholderData` produces an infinite loop. Used `initialData` instead and to imitate placeholder data, when the `user` value is null, we will fallback to the `initialUser`.

![Screenshot 2024-07-01 at 7 59 44 PM](https://github.com/dailydotdev/apps/assets/13744167/32dfa5dd-e69d-422c-aea0-fb8165960841)

Reply:
Probably good to debug a bit more then?
Feels like a re-render twice related thing?

Reply:
I tried making the parameters to the query to be stable, but none worked. Do you maybe have some ideas on where to look further?

Reply:
How to trigger this infinite loop? What are the repro steps?

Reply:
Just reload the page and it would do infinite loop if the `placeholderData` is used while viewing the page.